synapse/events/third_party_rules.py

```diff
@@ -217,7 +215,12 @@ async def check_event_allowed(
         event.freeze()
 
         for callback in self._check_event_allowed_callbacks:
-            res, replacement_data = await callback(event, state_events)
+            try:
+                res, replacement_data = await callback(event, state_events)
+            except Exception as e:
+                logger.warning("Failed to run callback %s: %s", callback, e)
```

Comment:
```suggestion
                logger.warning("Failed to run module API callback %s: %s", callback, e)
```

Just because "Failed to run callback" is pretty generic.